dot/sync/bootstrap_syncer.go

```diff
@@ -45,21 +50,22 @@ func (s *bootstrapSyncer) handleNewPeerState(ps *peerState) (*worker, error) {
 	}, nil
 }
 
-func (s *bootstrapSyncer) handleWorkerResult(res *worker) (*worker, error) {
+func (s *bootstrapSyncer) handleWorkerResult(res *worker) (
+	workerToRetry *worker, retry bool, err error) {
```

Comment:
whats the point of the `retry bool`? it seems if the workerToRetry isn't nil then retry is always true and same for the other way around

Reply:
It's just to satisfy the `nilnil` linter. It complains whenever we return an invalid result (`nil` worker) and a nil error.

I can revert it and ignore the linter for this particular case if you want though.

Reply:
Wellll that's up for debate. We can also use `//nolint:nilnil`, but the point of the linter is also to avoid any confusion, so returning an extra var like `retry bool` can sort of clarify things (although it's technically not necessary for it to work)

Reply:
I think it's cleaner in this case to just put a `//nolint` as it seems cleaner to me

Reply:
Reverted back in https://github.com/ChainSafe/gossamer/pull/2072/commits/de7f0973d2a9492f59cc349002fc4d75835f00d9